cleanlab_studio/cli/dataset/schema_helpers.py

```diff
@@ -164,19 +164,21 @@ def multiple_separate_words_detected(values: Collection[Any]) -> bool:
 
 
 def is_filepath(string: str, check_existing: bool = False) -> bool:
+    if pathlib.Path(string).suffix == "" or " " in string:
+        return False
     if check_existing:
         return pathlib.Path(string).exists()
-    return pathlib.Path(string).suffix != "" and " " not in string
+    return True
 
 
 def is_url(string: str, check_existing: bool = False) -> bool:
     try:
+        is_valid = validators.url(string)
+        assert isinstance(is_valid, bool)
         if check_existing:
             requests.head(string)
             return True
-        res = validators.url(string)
-        assert isinstance(res, bool)
-        return res
+        return is_valid
     except:
         return False
```

Comment:
we don't want all of this code in the try except block I think, just the head request. It looks like we're doing something kinda funky with using an assertion for the bool-ness of the return value. I'd recomend rewriting this bit for better readability

Reply:
```
if validators.url(string) is not True:
    return False
    
if check_existing:
    try:
        requests.head(string)
        return True
    except requests.RequestException:
        return False
        
return True
    
```

Reply:
So the `validators.url` function is kinda wonky. It returns `True` if the url is valid and throws an error if it's not valid. That's the reason for wrapping the whole thing in a try except block. I'll rewrite to get rid of the assert though

Reply:
based on their [docs](https://validators.readthedocs.io/en/latest/index.html#module-validators.url) and trying myself, it looks like it doesn't throw an error. it does return a `ValidationFailure` object that looks like an error however

agreed that it's a wonky API, but it shouldn't be needed to wrap that in the try except. it actually looks like the truthiness of `ValidationFailure` is false, so you can just do `if validators.url(string)...`)

Reply:
Ah yeah I just realized this haha. Assumed `ValidationFailure` was an error

Reply:
might be good to add some unit tests to these functions -- we can do this as a refinement task but these are pretty good candidates for testing, especially edge cases here

Reply:
I'll create a github issue for this